display_list/display_list.h

```diff
@@ -309,6 +311,8 @@ class DisplayList : public SkRefCnt {
     return modifies_transparent_black_;
   }
 
+  const DisplayListStorage& GetStorage() const { return storage_; }
```

Comment:
Is this a public method on DisplayList? What is it used for? The storage is supposed to be internal only.

Reply:
Now that storage is immutable there is no harm making it public.  This PR was pulled out from https://github.com/flutter/engine/pull/52715 which stores the display list.  It allows people to author tools that inspect the display list.

Reply:
The storage has always been immutable in practice (except for the one mutable field in it which will survive these changes because a mutable field is not disabled by a const cast) due to not being exposed. Exposing the buffer as a const pointer would not require such extensive changes and would have had the same external effect. But, I don't want to see this exposed in any case, because...

The harm by making it public is to invite the concept that the data within it is meaningful in any way outside of the DL implementation itself. The implementation can and does change frequently, so the stream of data is not meaningful in any lasting (or even temporary) manner. By exposing it, it's as if we're promising reusability or meaningfulness for the data that is exposed and both of those are (while not stated in the doc comments) deliberately not supported.

This is not a substitute for a serialization/readback mechanism and isn't really useful as a manual inspection tool especially in light of the fact that there exists a supported pretty printing tool for DisplayLists in https://github.com/flutter/engine/blob/900ca9a186d3dabaa8fd5afd1fe93e9e7bcb558d/testing/display_list_testing.h#L33